### PR TITLE
Fix tests.

### PR DESCRIPTION
**Pull request by SpacePossum**
somehow the PHP check of the fixer itself must be bypassed (https://travis-ci.org/keradus/PHP-CS-Fixer/jobs/203770470#L292) or it should not run as phar on the 7.2/nightly test suit.

### CODE REVIEW

**Comment by SpacePossum** [comment]
@keradus 
https://travis-ci.org/keradus/PHP-CS-Fixer/jobs/205701196#L915
interesting failure, I wonder if we should call `next()` first... maybe this is a odd diff. between PHP/Zend and HHVM?

File: .travis.yml [hunk 1]
```diff
@@ -82,8 +82,7 @@ script:
     - if [ $TASK_TESTS == 1 ] && [ $TASK_TESTS_COVERAGE == 0 ]; then vendor/bin/phpunit --verbose; fi
     - if [ $TASK_TESTS == 1 ] && [ $TASK_TESTS_COVERAGE == 1 ]; then phpdbg -qrr vendor/bin/phpunit --verbose --coverage-clover build/logs/clover.xml; fi
 
-    - if [ $TASK_SCA == 1 ]; then git checkout . -q; fi
-
+    - if [ $TASK_CS == 1 ]; then git checkout . -q; fi
```
Comment (keradus) [inline .travis.yml]: why this change? if no SCA was performed, there is no files to checkout, isn't it ?

Comment (SpacePossum) [inline .travis.yml]: ah yes, this change is not good

File: .travis.yml [hunk 2]
```diff
@@ -21,7 +21,7 @@ matrix:
         - php: 7.1
           env: TASK_SCA=1
         - php: nightly
-          env: TASK_SCA=1 COMPOSER_FLAGS="--ignore-platform-reqs"
+          env: COMPOSER_FLAGS="--ignore-platform-reqs" TASK_CS=0
```
Comment (keradus) [inline .travis.yml]: what I was thinking actually was introduction of new ENV flag:
`PHP_CS_FIXER_IGNORE_ENV=1` and **allow** tool to run on 7.2 with that flag.
For that, we would be able to run the SCA

File: tests/ConfigTest.php [hunk 3]
```diff
@@ -107,7 +107,7 @@ public function testThatFinderWorksWithDirSetOnConfig()
         $config = new Config();
 
         $iterator = $config->getFinder()->in(__DIR__.'/Fixtures/FinderDirectory')->getIterator();
-        $this->assertSame(1, count($iterator));
+        $this->assertSame(1, iterator_count($iterator));
```
Comment (keradus) [inline tests/ConfigTest.php]: interesting, do you have an ref for that change ?
would `assertCount` finally works ?

**Comment by keradus** [comment]
why calling `next` would help? then, we would drop one value from collection, wound't we ?

File: tests/ConfigTest.php (hunk 3 above)
Comment (SpacePossum) [inline tests/ConfigTest.php]: [`assertCount`](https://github.com/sebastianbergmann/phpunit/blob/4.0/src/Framework/Constraint/Count.php#L105) will work I think, however we might run in the old HHVM bug

**Comment by SpacePossum** [comment]
indeed, calling `next` is wrong, I'm just puzzled why this fails on HHVM and PHP
this looks OK (https://3v4l.org/CEUWA), so maybe it is in the iterator itself.

File: tests/ConfigTest.php (hunk 3 above)
Comment (keradus) [inline tests/ConfigTest.php]: yeah, my question is more like "is there a hhvm version that has fixed that bug already" :D

Comment (keradus) [inline tests/ConfigTest.php]: so, it still not working on hhvm :/

**Comment by SpacePossum** [comment]
My guess is https://3v4l.org/ZjdMV i.e. something like HHVM and RecursiveRegexIterator for ref.: https://github.com/facebook/hhvm/issues/4966

File: tests/ConfigTest.php [hunk 4]
```diff
@@ -133,7 +133,7 @@ public function testThatCustomSymfonyFinderWorks()
         $config = Config::create()->setFinder($finder);
 
         $iterator = $config->getFinder()->getIterator();
-        $this->assertSame(1, count($iterator));
+        $this->assertCount(1, $iterator);
```
Comment (keradus) [inline tests/ConfigTest.php]: @SpacePossum , fix to make it working properly on hhvm is to run `iterator_to_array($iterator)` first :/
still, even hhvm 3.18 is not fixed that bug... what a shame for hhvm...

Comment (SpacePossum) [inline tests/ConfigTest.php]: yeah :(
Wonder if HHVM is not hit by same bug as PHP was;
https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/Finder/Iterator/FilterIterator.php#L32

Comment (keradus) [inline tests/ConfigTest.php]: wooow, so great :/

**Comment by keradus** [comment]
@SpacePossum , could you review it please: https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2552 ?
Travis is green for both 2.1 and master lines
